src/map/script.c

```diff
@@ -21025,6 +21025,37 @@ static BUILDIN(unitwalk)
 	return true;
 }
 
+/**
+ * Checks if a unit is walking.
+ *
+ * Returns 1 if unit is walking, 0 if unit is not walking and -1 on error.
+ *
+ * @code{.herc}
+ *	unitiswalking(<GID>);
+ * @endcode
+ *
+ **/
+static BUILDIN(unitiswalking)
+{
+	struct block_list *bl = map->id2bl(script_getnum(st, 2));
```

Comment:
parameter probably can be optional.
if parameter missing, you may use ``st->oid``

Reply:
Done.